doom-themes-ext-treemacs.el

```diff
@@ -94,81 +98,310 @@ variable-pitch face."
   (treemacs-create-theme "doom"
```

Comment:
Perhaps it should be two separate treemacs themes?

Reply:
I don't know, what do you think? How would you do it?

Reply:
Like maybe in a separate file `doom-themes-ext-treemacs2.el` with a `(doom-themes-treemacs-config2)`?

Reply:
I'm in favor of either, the current way makes it easy for the user to flip it on and off though, and not enable both.

Reply:
Yeah, maybe it's better. The only way it could affect is for future themes, but I'm not quite sure if that makes sense (to implement more Treemacs themes)

Reply:
I was just thinking two calls to `treemacs-create-theme`, making separate `doom-atom` and `doom-colors` themes, and changing the last line to `(treemacs-load-theme (if doom-themes-treemacs-color-icons "doom-atom" "doom-colors")`.  They don't need to live in separate files.

Just a suggestion, I'm not strongly opposed to how it is now, just thinking about how best to organise it.

Reply:
Ok, I think what you say makes sense, the thing is that they're both the same theme but it's just the code file icons that change, but what you say it's right. Let me figure it out

Reply:
Otherwise if it's a variant of the same theme, I think the conditional scope should be reduce so we're not duplicating definitions on unaffected icons like "repo" and "tag".

Reply:
There's indeed some redundancy. Thanks for the heads up. The tag does change, but there's some thing that doesn't like the media icons and some other stuff. Let's make a decision, shall I separate them in two themes or leave it as a variant?

Reply:
Perhaps variant is better, at least it will be shorter if the duplication is resolved.

Reply:
Would it be helpful to maybe abstract out so that anyone could make a treemacs theme?

Reply:
@Emiller88 maybe, but the effort would be too much given that I don't see another Treemacs theme implementation coming anytime soon

Reply:
Removed redundancy. I think it's ready to merge but please triple check. I'm constantly using it to test it and until now I haven't found errors.